assets/js/googlesitekit/datastore/user/date-range.test.js

```diff
@@ -44,6 +45,21 @@ describe( 'core/user date-range', () => {
 				expect( registry.select( STORE_NAME ).getDateRange() ).toEqual( someDateRange );
 			} );
 		} );
+
+		describe( 'setReferenceDate', () => {
+			it( 'should require the date string param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setReferenceDate();
+				} ).toThrow( 'Date string is required.' );
+			} );
+
+			it( 'should set the reference date', () => {
+				const someReferenceDate = '2020-09-12';
+
+				registry.dispatch( STORE_NAME ).setReferenceDate( someReferenceDate );
+				expect( registry.select( STORE_NAME ).getReferenceDate() ).toEqual( someReferenceDate );
```

Comment:
Generally I prefer avoiding testing the state directly in tests, but for most actions I think it's preferable to avoid mixing potential logic in the selector, which has its own tests – especially since we usually need to use the action in the selector test.

```suggestion
				expect( registry.stores[ STORE_NAME ].store.getState().referenceDate ).toEqual( someReferenceDate );
```